pkg/operator/ceph/cluster/key_rotation.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2025 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package cluster to manage a Ceph cluster.
+package cluster
+
+import (
+	"github.com/pkg/errors"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/daemon/ceph/client"
+	"github.com/rook/rook/pkg/operator/ceph/config"
+	"github.com/rook/rook/pkg/operator/ceph/config/keyring"
+	"github.com/rook/rook/pkg/operator/ceph/reporting"
+)
+
+// ClusterKeysToRotate defines the type of ceph daemon keys that need rotation
+type ClusterKeysToRotate struct {
+	RBDMirrorPeer bool
+}
+
+// GetKeysToRotate returns the KeysToRotate struct that defines the type of keys that need rotation.
+func (c *cluster) GetKeysToRotate() (ClusterKeysToRotate, error) {
+	keysToRotate := ClusterKeysToRotate{}
+	cephObj := &cephv1.CephCluster{}
+	if err := c.context.Client.Get(c.ClusterInfo.Context, c.namespacedName, cephObj); err != nil {
+		return keysToRotate, errors.Wrapf(err, "failed to get cluster %v to get the cephx keys to rotate.", c.namespacedName)
+	}
+
+	// cephX status won't be initialized for a new cluster.
+	if cephObj.Status.CephX == nil {
+		return keysToRotate, nil
+	}
+
+	desiredCephVersion := c.ClusterInfo.CephVersion
+	// check the least version for OSD since mons might have been upgraded by now.
+	runningCephVersion, err := client.LeastUptodateDaemonVersion(c.context, c.ClusterInfo, config.MonType)
+	if err != nil {
+		return keysToRotate, errors.Wrapf(err, "failed to get the lowest running ceph version in the cluster %v.", c.namespacedName)
+	}
+
+	allCephDaemonTypes := []string{client.RBDMirrorPeerKeyringID}
+	for _, daemon := range allCephDaemonTypes {
+		switch daemon {
+		case client.RBDMirrorPeerKeyringID:
+			shouldRotateKeys, err := keyring.ShouldRotateCephxKeys(cephObj.Spec.Security.CephX.RBDMirrorPeer, runningCephVersion, desiredCephVersion, *cephObj.Status.CephX.RBDMirrorPeer)
+			if err != nil {
+				return keysToRotate, errors.Wrapf(err, "failed to check if %q daemon keys should be rotated or not", daemon)
+			}
+			keysToRotate.RBDMirrorPeer = shouldRotateKeys
+		}
+	}
+
+	logger.Debugf("ceph daemon keys to rotate in cluster %v are: %v", c.namespacedName, keysToRotate)
+	return keysToRotate, nil
+}
+
+// UpdateClusterCephxStatus fetches the latest cephCluster instance and updates cephxStatus for a particular ceph daemon
+func (c *cluster) UpdateClusterCephxStatus(cephxDaemonType string, didRotate bool) error {
```

Comment:
The problem we are going to run into with this method is that CSI needs to report additional information on top of `UpdatedCephxStatus()`, which won't match this function signature, and I can't think of a good way to make it match.

For now, let's leave status updates to each subsection of the CephCluster reconcile doing the rotation. Once we have all implementations completed, it will be easier to see what we can condense to keep code DRY and avoid having individual subsections of hte reconcile diverge from established pattern.